src/main/kotlin/frc/robot/Initializer.kt

```diff
@@ -22,31 +11,10 @@ import org.photonvision.PhotonCamera
 object Initializer {
 
     fun initSwerve() {
-        var moduleIOs: Array<ModuleIO>
+        val moduleIOs: Array<ModuleIO>
 
-        when (SwerveConstants.SWERVE_TYPE) {
-            SwerveConstants.SwerveType.SIM -> {
-                moduleIOs = Array<ModuleIO>(4) { ModuleIOSim() }
-                SwerveDrive.initialize(GyroIOSim(), SwerveConstants.OFFSETS, *moduleIOs)
-            }
-
-            SwerveConstants.SwerveType.WCP -> {
-                moduleIOs = Array<ModuleIO>(4) { i ->
-                    ModuleIOTalonFX(
-                        Ports.SwerveDriveWCP.DRIVE_IDS[i],
-                        Ports.SwerveDriveWCP.ANGLE_IDS[i],
-                        Ports.SwerveDriveWCP.ENCODER_IDS[i],
-                        SwerveConstants.DRIVE_MOTOR_CONFIGS
-                            ?: throw IllegalStateException("drive motor config is null"),
-                        SwerveConstants.ANGLE_MOTOR_CONFIGS
-                            ?: throw IllegalStateException("angle motor config is null"),
-                        SwerveConstants.ENCODER_CONFIGS ?: throw IllegalStateException("encoder config is null")
-                    )
-                }
-                SwerveDrive.initialize(GyroIOReal(), SwerveConstants.OFFSETS, *moduleIOs)
-            }
-
-            SwerveConstants.SwerveType.NEO -> {
+        if (Constants.CURRENT_MODE == Constants.Mode.REAL) {
+            if (Constants.ROBORIO_SERIAL_NUM == Constants.ROBORIO_NEO_SERIAL) {
```

Comment:
Perhaps creating a map between the robot type and the needed modules will be better.

Reply:
73b6862e8840e941e184cf76bdb3582dc82cb4ae